generated/source-repo/githubactions/.github/workflows/build-and-update-gitops.yml

```diff
@@ -9,21 +9,23 @@
 name: TSSC-Build-Attest-Scan-Deploy
 env:
   CI_TYPE: github
+
   # 🖊️ EDIT to change the image registry settings.
   # Registries such as GHCR, Quay.io, and Docker Hub are supported.
   IMAGE_REGISTRY: ${{ secrets.IMAGE_REGISTRY }}
   IMAGE_REGISTRY_USER: ${{ secrets.IMAGE_REGISTRY_USER }}
   IMAGE_REGISTRY_PASSWORD: ${{ secrets.IMAGE_REGISTRY_PASSWORD }}
 
-  ROX_CENTRAL_ENDPOINT: ${{ secrets.ROX_CENTRAL_ENDPOINT }}
   ROX_API_TOKEN: ${{ secrets.ROX_API_TOKEN }}
-
+  ROX_CENTRAL_ENDPOINT: ${{ secrets.ROX_CENTRAL_ENDPOINT }}
+  GITOPS_AUTH_PASSWORD: ${{ secrets.GITOPS_AUTH_PASSWORD }}
+  # Uncomment this when using Gitlab
+  # GITOPS_AUTH_USERNAME: ${{ secrets.GITOPS_AUTH_USERNAME }}
```

Comment:
It's silly to include this here, so I'll think of some exclusion mechanism (maybe just ignore this one specific secret directly in the template code)

Reply:
> It's silly to include this here, so I'll think of some exclusion mechanism (maybe just ignore this one specific secret directly in the template code) 

I was thinking we automatically include if the repo is gitlab ... Its kind of a pain to have to uncomment it... We can  fix automate in the generated templates 
See this https://github.com/redhat-appstudio/tssc-sample-templates/blob/main/skeleton/gitops-template/components/http/overlays/development/source-repository.yaml  as an example
In this repo, we have to generate this code in the jenkinsfile for use in the tssc-sample-template ( template generating code to be expanded later( 
    {%- if values.hostType == 'Gitlab' %} 
    GITOPS_AUTH_USERNAME: ${{ secrets.GITOPS_AUTH_USERNAME }}
    {%- endif %}

Reply:
This is the GH actions pipeline, so the repo is never Gitlab (which is what I meant by "it's silly to include this")

Reply:
> This is the GH actions pipeline, so the repo is never Gitlab (which is what I meant by "it's silly to include this")

Agree.
We're talking about a two things.
1. automatically adding the line if the repo the jenkinsfile is in fact  a gitlab repo. We can do that at template generation time when we know the destination git type and the user doen't need to undo the commenting out 
2. Your point that Github should not have it - the mechanism for that in the generator is fine but the case (1) would take care of it but I agree would look silly

We should do both